apps/rush/src/utilities/Telemetry.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import * as path from 'path';
+import * as fsx from 'fs-extra';
+
+import {
+  RushConfiguration
+} from '@microsoft/rush-lib';
+
+export interface ITelemetryData {
+  name: string;
+  duration: number;
+  result: string;
+  extraData?: { [key: string]: string };
+}
+
+export default class Telemetry {
+  private _enabled: boolean;
+  private _store: ITelemetryData[];
+  private _dataFolder: string;
+
+  public constructor(private _rushConfiguration: RushConfiguration) {
```

Comment:
It is not for sharing yet. 

---
In reply to: [117845857](https://github.com/Microsoft/web-build-tools/pull/207#discussion_r117845857) [](ancestors = 117845857)